### PR TITLE
Some more about SASS printing/checking

### PR DESCRIPTION
**Pull request by zasdfgbnm**
- Add debug print option `sass`, which will print the disassembled SASS string into stdout
- Some test code refactoring
- Add basic sanity check for nvFuser matmul kernel's SASS

### CODE REVIEW

File: third_party/nvfuser/test/test_gpu_external_src.cpp [hunk 1]
```diff
@@ -85,51 +85,6 @@ TEST_F(NVFuserTest, FusionExternalSrc_CUDA) {
   }
 }
 
-namespace {
```
Comment (zasdfgbnm) [inline third_party/nvfuser/test/test_gpu_external_src.cpp:88]: Remove copy-pasted code

File: third_party/nvfuser/test/test_gpu_tensorcore.cpp [hunk 2]
```diff
@@ -40,127 +40,11 @@
...
 
 // Tests go in torch::jit
-namespace torch {
-namespace jit {
+namespace torch::jit {
 
 using namespace torch::jit::fuser::cuda;
 using namespace at::indexing;
 
-namespace {
```
Comment (zasdfgbnm) [inline third_party/nvfuser/test/test_gpu_tensorcore.cpp:49]: Moved to `test_utils.{h,cpp}`

File: third_party/nvfuser/test/test_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,275 @@
+#include <test/test_utils.h>
```
Comment (zasdfgbnm) [inline third_party/nvfuser/test/test_utils.cpp:1]: Mostly code movement. The only new thing is the `namespace sass`

File: third_party/nvfuser/test/test_utils.cpp [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    actual_mismatched_snippet = actual_mismatched_snippet.substr(0, 10);
+    std::string expected_mismatched_snippet(it.first, expected_kernel.end());
+    expected_mismatched_snippet = expected_mismatched_snippet.substr(0, 10);
+    std::cerr << "First mismatch found at: " << actual_mismatched_snippet
+              << ", expected: " << expected_mismatched_snippet << std::endl;
+    TORCH_CHECK(false);
+  }
+}
+
+namespace sass {
```
Comment (zasdfgbnm) [inline third_party/nvfuser/test/test_utils.cpp:61]: Everything in this namespace is new

File: third_party/nvfuser/test/test_utils.h [hunk 5]
```diff
@@ -395,32 +378,94 @@ class NVFuserTest : public ::testing::Test {
 };
 
 // assert that the given fusion lowers to the given CUDA kernel
-inline void assertCUDAKernel(
```
Comment (zasdfgbnm) [inline third_party/nvfuser/test/test_utils.h:398]: Code moved to `test_utils.cpp`

File: third_party/nvfuser/test/test_utils.h [hunk 6]
```diff
@@ -395,32 +378,94 @@ class NVFuserTest : public ::testing::Test {
...
-    std::string actual_mismatched_snippet(it.second, actual_kernel.end());
-    actual_mismatched_snippet = actual_mismatched_snippet.substr(0, 10);
-    std::string expected_mismatched_snippet(it.first, expected_kernel.end());
-    expected_mismatched_snippet = expected_mismatched_snippet.substr(0, 10);
-    std::cerr << "First mismatch found at: " << actual_mismatched_snippet
-              << ", expected: " << expected_mismatched_snippet << std::endl;
-    TORCH_CHECK(false);
+void assertCUDAKernel(Fusion* fusion, const std::string& expected_kernel);
+
+namespace sass {
```
Comment (zasdfgbnm) [inline third_party/nvfuser/test/test_utils.h:383]: This namespace is newly added.

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**